setup/flex.rst

```diff
@@ -0,0 +1,173 @@
+enabled automatically (``all`` in this case). The ``env`` option makes Flex to
+add new environment variables to the application. Finally, the ``copy-from-recipe``
+option allows the recipe to copy files and directories into your application.
+
+The instructions defined in this ``manifest.json`` file are also used by Symfony
+Flex when uninstalling dependencies (e.g. ``composer remove mailer``) to undo
+all changes. This means that Flex can remove the SwiftmailerBundle from the
+application, delete the ``MAILER_URL`` environment variable and any other file
+and directory created by this recipe.
+
+Symfony Flex recipes are contributed by the community and they are stored in
+two public repositories:
+
+* `Main recipe repository`_, is a curated list of recipes for high quality and
+  maintained packages. Symfony Flex only looks for in this repository by default.
+* `Contrib recipe repository`_, contains all the recipes created by the community.
+  All of them are guaranteed to work, but their associated packages could be
+  unmaintained. Symfony Flex ignores these recipes by default, but you can execute
+  this command to start using them in your project:
+
+  .. code-block:: terminal
+
+        $ cd your-project/
+        $ composer config extra.symfony.allow-contrib true
+
+Read the `Symfony Recipes documentation`_ to learn everything about how to
+created recipes for your own packages.
+
+Using Symfony Flex in New Applications
+--------------------------------------
+
+Symfony has published a new "skeleton" project, which is a minimal Symfony
+project recommended to create new applications. This "skeleton" already includes
+Symfony Flex as a dependency, so you can create a new Flex-enabled Symfony
+application executing the following command:
+
+.. code-block:: terminal
+
+    $ composer create-project symfony/skeleton my-project
+
+.. note::
+
+    The use of the Symfony Installer to create new applications is no longer
+    recommended since Symfony 3.3. Use Composer ``create-project`` command instead.
+
+Upgrading Existing Applications to Flex
+---------------------------------------
+
+Using Symfony Flex is optional, even in Symfony 4, where Flex will be used by
+default. However, Flex is so convenient and improves your productivity so much
+that it's strongly recommended to upgrade your existing applications to it.
+
+The only caveat is that Symfony Flex requires that applications use the
+following directory structure, which is the same used by default in Symfony 4:
+
+.. code-block:: text
+
+    your-project/
+    ├── Makefile
+    ├── config/
+    │   ├── bundles.php
+    │   ├── container.yaml
+    │   ├── packages/
+    │   └── routing.yaml
+    ├── public/
+    │   └── index.php
+    ├── src/
+    │   ├── ...
+    │   └── Kernel.php
+    ├── templates/
+    └── vendor/
+
+This means that installing the ``symfony/flex`` dependency in your application
+is not enough. You must also upgrade the directory structure to the one showed
```

Comment:
Shouldn't we explain the options you can use in your `composer.json` file to adapt for alternative directory names instead?